File: 백준/Silver/17615. 볼 모으기/볼 모으기.swift
import Foundation

let N = Int(readLine()!)!
var arr = readLine()!.map{String($0)}
var ans = N
var flag = false
var res = 0

for i in 0..<N{
    if arr[i] == "R"{
        if flag {res += 1}
    }else{
        flag = true
    }
}
ans = min(ans, res)
flag = false
res = 0

for i in stride(from: N-1, through: 0, by: -1){
    if arr[i] == "R"{
        if flag {res += 1}
    }else{
        flag = true
    }
}
ans = min(ans, res)
flag = false
res = 0

for i in 0..<N{
    if arr[i] == "B"{
        if flag {res += 1}
    }else{
        flag = true
    }
}
ans = min(ans, res)
flag = false
res = 0

for i in stride(from: N-1, through: 0, by: -1){
    if arr[i] == "B"{
        if flag {res += 1}
    }else{
        flag = true
    }
}
ans = min(ans, res)
flag = false
res = 0

print(ans)

File: 백준/Silver/17615. 볼 모으기/볼 모으기.cc
#include <iostream>
#include <vector>

using namespace std;

int main(int argc, const char * argv[]) {
    int N, ans;
    vector<char> arr;
    
    cin >> N;
    ans = N;
    for (int i=0; i<N; i++){
        char input;
        cin >> input;
        arr.push_back(input);
    }
    int res = 0;
    bool flag = false;
    
    for (int i=0; i<N; i++){
        if(arr[i]=='R'){
            res += flag ? 1:0;
        }else{
            flag = true;
        }
    }
    ans = min(ans,res);
    res = 0;
    flag = false;
    
    for (int i=N-1; i>=0; i--){
        if(arr[i]=='R'){
            res += flag ? 1:0;
        }else{
            flag = true;
        }
    }
    ans = min(ans,res);
    res = 0;
    flag = false;

    for (int i=0; i<N; i++){
        if(arr[i]=='B'){
            res += flag ? 1:0;
        }else{
            flag = true;
        }
    }
    ans = min(ans,res);
    res = 0;
    flag = false;
    
    for (int i=N-1; i>=0; i--){
        if(arr[i]=='B'){
            res += flag ? 1:0;
        }else{
            flag = true;
        }
    }
    ans = min(ans,res);
    res = 0;
    flag = false;
    cout << ans << "\n";
    return 0;
}